### ISSUE TITLE
JobDSL does not work with current configuration for hpi:run

### ISSUE THREAD

**Issue by MadsNielsen** [Praqma/pretested-integration-plugin#102]
The current pom has an optional dependency to a version of JobDSL that requires a newer version of Jenkins core than the one specified. We require at least 2.19.4.

```
WARNING: Failed to instantiate Key[type=javaposse.jobdsl.plugin.actions.ApiViewerRootAction, annotation=[none]]; skipping this component
com.google.inject.ProvisionException: Guice provision errors:

1) Error injecting constructor, java.lang.NoClassDefFoundError: org/codehaus/groovy/runtime/typehandling/ShortTypeHandling
  at javaposse.jobdsl.plugin.actions.ApiViewerRootAction.<init>(ApiViewerRootAction.groovy:28)

1 error
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:52)
	at com.google.inject.Scopes$1$1.get(Scopes.java:65)
	at hudson.ExtensionFinder$GuiceFinder$FaultTolerantScope$1.get(ExtensionFinder.java:427)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at com.google.inject.internal.InjectorImpl$3$1.call(InjectorImpl.java:1005)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1051)
	at com.google.inject.internal.InjectorImpl$3.get(InjectorImpl.java:1001)
	at hudson.ExtensionFinder$GuiceFinder._find(ExtensionFinder.java:389)
	at hudson.ExtensionFinder$GuiceFinder.find(ExtensionFinder.java:380)
	at hudson.ClassicPluginStrategy.findComponents(ClassicPluginStrategy.java:388)
	at hudson.ExtensionList.load(ExtensionList.java:349)
	at hudson.ExtensionList.ensureLoaded(ExtensionList.java:287)
	at hudson.ExtensionList.iterator(ExtensionList.java:156)
	at jenkins.model.Jenkins$20.run(Jenkins.java:2789)
	at org.jvnet.hudson.reactor.TaskGraphBuilder$TaskImpl.run(TaskGraphBuilder.java:169)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:282)
	at jenkins.model.Jenkins$8.runTask(Jenkins.java:924)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:210)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NoClassDefFoundError: org/codehaus/groovy/runtime/typehandling/ShortTypeHandling
 ```

**Comment by MadsNielsen** [comment]
Solved in #103

**Comment by buep** [comment]
This issue came from our effort of writing a test using job dsl snippets in #64, but where @MadsNielsen experienced the problem that the job dsl plugin we have as a dependency in our pretested integration plugin requires a never core. We need the dependency to job dsl in the current version so we need to update core and what-else comes after that change.
